eks/cleanup.go

```diff
@@ -0,0 +1,311 @@
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("group-id"),
+				Values: []*string{aws.String(securityGroupID)},
+			},
+		},
+	}
+	networkInterfacesResult, err := ec2Svc.DescribeNetworkInterfaces(describeNetworkInterfacesInput)
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+	for _, ni := range networkInterfacesResult.NetworkInterfaces {
+		logger.Infof("Found network interface %s", aws.StringValue(ni.NetworkInterfaceId))
+	}
+
+	// Detach network interfaces
+	for _, ni := range networkInterfacesResult.NetworkInterfaces {
+		detachInput := &ec2.DetachNetworkInterfaceInput{
+			AttachmentId: ni.Attachment.AttachmentId,
+		}
+		_, err := ec2Svc.DetachNetworkInterface(detachInput)
+
+		// If we have an error, check that it's NotFound. This is actually a success.
+		if err != nil {
+			if awsErr, isAwsErr := err.(awserr.Error); isAwsErr && awsErr.Code() == "InvalidAttachmentID.NotFound" {
+				logger.Infof("Network interface %s is detached.", aws.StringValue(ni.NetworkInterfaceId))
+			}
+			return errors.WithStackTrace(err)
+		}
+		logger.Infof("Requested to detach network interface %s for security group %s", aws.StringValue(ni.NetworkInterfaceId), securityGroupID)
+	}
+
+	// Wait for network interfaces to be detached
+	if len(networkInterfacesResult.NetworkInterfaces) > 0 {
+		err = waitForNetworkInterfacesToBeDetached(ec2Svc, networkInterfacesResult.NetworkInterfaces, waitMaxRetries, waitSleepBetweenRetries)
+		if err != nil {
+			return err
+		}
+		logger.Info("Verified network interfaces are detached.")
+	}
+
+	// Delete network interfaces
+	for _, ni := range networkInterfacesResult.NetworkInterfaces {
+		deleteNetworkInterfacesInput := &ec2.DeleteNetworkInterfaceInput{
+			NetworkInterfaceId: ni.NetworkInterfaceId,
+		}
+		_, err := ec2Svc.DeleteNetworkInterface(deleteNetworkInterfacesInput)
+
+		if err != nil {
+			return errors.WithStackTrace(err)
+		}
+		logger.Infof("Requested to delete network interface %s for security group %s", aws.StringValue(ni.NetworkInterfaceId), securityGroupID)
+	}
+
+	// Wait for network interfaces to be deleted
+	err = waitForNetworkInterfacesToBeDeleted(ec2Svc, networkInterfacesResult.NetworkInterfaces, waitMaxRetries, waitSleepBetweenRetries)
+	if err != nil {
+		return err
+	}
+	logger.Info("Verified network interfaces are deleted.")
+
+	return nil
+}
+
+// Used to look up the security group for the ALB ingress controller
+func lookupSecurityGroup(
+	ec2Svc *ec2.EC2,
+	vpcID string,
+	clusterID string,
+) (*ec2.DescribeSecurityGroupsOutput, error) {
+	logger := logging.GetProjectLogger()
+
+	logger.Infof("Looking up security group containing tag for EKS cluster %s", clusterID)
+	sgInput := &ec2.DescribeSecurityGroupsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{aws.String(vpcID)},
+			},
+			{
+				Name:   aws.String("tag:kubernetes.io/cluster-name"),
+				Values: []*string{aws.String(clusterID)},
+			},
+		}}
```

Comment:
This needs the other kind of tag filter, with `elbv2.k8s.aws/cluster`. We can add that after this is merged, but would be useful to either have a TODO marker or github issue so we don't forget.

Reply:
I wasn't able to find a leftover SG with this tag: `kubernetes.io/cluster-name`, so I'm not sure if we still need that lookup? I do find the `elbv2.k8s...` tag, and that's now getting cleaned up.